R/Combine_and_Normalize.R

```diff
@@ -502,73 +558,74 @@ combineNormalize <- function(object,
     
     k <- 1
     pca.grob=list()
-  for (i in 1:length(vars.to.regress)){
-    v=vars.to.regress[i]
-    print(v)
-    r <- lapply(names(object.merge.split), 
-                function(x){.plotPCA(object.merge.split[[x]],v,x)})
-    nr <- lapply(names(object.merge.nr.split), 
-                 function(x){.plotPCA(object.merge.nr.split[[x]],v,x)})
-
-    grob = ggarrange(plotlist=r,ncol=1,
-                     common.legend = F,
-                     legend = 'right')%>%
-      annotate_figure(top=text_grob(paste0(v,' Regression'), 
-                                    face = "bold", size = 20),
-                      fig.lab.size = 20,
-                      fig.lab.face = 'bold',
-                      fig.lab.pos='top')    
-    grob.nr = ggarrange(plotlist=nr
-                        ,ncol=1,
-                        common.legend = F,
-                        legend = 'right')%>%
-      annotate_figure(top=text_grob('No Regression', 
-                                    face = "bold", size = 20),
-                      fig.lab.size = 20,
-                      fig.lab.face ='bold',
-                      fig.lab.pos='top')
-    
-    pca.grob=ggarrange(grob,grob.nr)
-  
-    grobsList[['Regression Plots']][[v]] =pca.grob
+    for (i in 1:length(vars.to.regress)){
+      v=vars.to.regress[i]
+      print(v)
+      r <- lapply(names(object.merge.split), 
+                  function(x){.plotPCA(object.merge.split[[x]],v,x)})
+      nr <- lapply(names(object.merge.nr.split), 
+                   function(x){.plotPCA(object.merge.nr.split[[x]],v,x)})
+      
+      grob = ggarrange(plotlist=r,ncol=1,
+                       common.legend = F,
+                       legend = 'right')%>%
+        annotate_figure(top=text_grob(paste0(v,' Regression'), 
+                                      face = "bold", size = 20),
+                        fig.lab.size = 20,
+                        fig.lab.face = 'bold',
+                        fig.lab.pos='top')    
+      grob.nr = ggarrange(plotlist=nr
+                          ,ncol=1,
+                          common.legend = F,
+                          legend = 'right')%>%
+        annotate_figure(top=text_grob('No Regression', 
+                                      face = "bold", size = 20),
+                        fig.lab.size = 20,
+                        fig.lab.face ='bold',
+                        fig.lab.pos='top')
+      
+      pca.grob=ggarrange(grob,grob.nr)
+      
+      grobsList[['Regression Plots']][[v]] =pca.grob
+      
+    }
     
-  }
   } else {
-    print("Regression not prefomred")
+    print("PCA regression plots not created")
   }
   
+  ### Determin # of PCs ====
+    #### create Elbow plot ====
+  if ('none'%in%methods.pca==F) {
```

Comment:
Not sure what this means.

Reply:
Methods can be multiple selections, I am tying to prioritize not running this section so if on biowulf user selects all options it will stop run

Reply:
I've made changes to the code: parameter choice for "none" changed to NULL.

Reply:
@bianjh-cloud Adding you to the conversation